### PR TITLE
add an experiment to prefer TLS 1.3 over secio

### PR DESCRIPTION
**Pull request by marten-seemann**
Depends on https://github.com/ipfs/go-ipfs-config/pull/32.

The CI build is expected to fail until this PR is merged and a release is tagged, which I can then import from this PR.

### CODE REVIEW

**Comment by Stebalien** [comment]
This will need to be reworked a bit for the constructor changes we just merged. Want me to handle that?

**Comment by marten-seemann** [comment]
That would be great. Thank you!

File: go.mod [hunk 1]
```diff
@@ -122,3 +124,5 @@ require (
 )
 
 go 1.12
+
+replace github.com/ipfs/go-ipfs-config => github.com/marten-seemann/go-ipfs-config v0.2.5-0.20190418001109-c95be0eacc9f
```
Comment (Stebalien) [inline go.mod]: @marten-seemann FYI, you can test PRs that depend on other PRs by adding replace statements.

Comment (marten-seemann) [reply go.mod]: I'm aware of that, but I didn't want to put it in a PR, because we might merge it accidentally without removing the `replace` first.

Comment (Stebalien) [reply go.mod]: I wonder if there's a bot we could employ for that.

**Comment by Stebalien** [comment]
I'll bring this up in the standup tomorrow. Assuming nobody objects, we can go ahead and merge it.

**Comment by Stebalien** [comment]
Actually, I still want a review from @magik6k on the constructor stuff.

File: docs/experimental-features.md [hunk 2]
```diff
@@ -683,3 +683,22 @@ ipfs config --json Swarm.EnableAutoNATService true
 ### Road to being a real feature
 
 - [ ] needs testing
+
+
+## TLS 1.3 as default handshake protocol
+
+### State
+
+Every node accepts secio and TLS 1.3 connections, but prefers secio over TLS when dialing.
```
Comment (magik6k) [inline docs/experimental-features.md]:
```suggestion
Every node accepts secio and TLS 1.3 connections, but prefers TLS over secio when dialing.
```

**Comment by Stebalien** [comment]
:bomb:

Thanks @marten-seemann!

**Pull request merged**